sword-server/src/main/java/edu/unc/lib/dl/cdr/sword/server/managers/AbstractFedoraManager.java

```diff
@@ -56,20 +62,15 @@ protected PID extractPID(String uri, String basePath) {
         if (pidString.trim().length() == 0) {
             targetPID = RepositoryPaths.getContentRootPid();
         } else {
-            targetPID = new DatastreamPID(pidString);
+            targetPID = PIDs.get(pidString);
         }
         return targetPID;
     }
 
-    protected boolean hasAccess(AuthCredentials auth, PID pid, Permission permission, SwordConfigurationImpl config) {
-        if (config.getAdminDepositor() != null && config.getAdminDepositor().equals(auth.getUsername())) {
-            return true;
+    protected void assertHasAccess(String message, PID pid, Permission permission) throws SwordError {
+        if (!aclService.hasAccess(pid, getAgentPrincipals().getPrincipals(), permission)) {
+            throw new SwordError(ErrorURIRegistry.INSUFFICIENT_PRIVILEGES, 403, message);
```

Comment:
would you mind using Response.Status.Forbidden here?

Reply:
Good catch, i've updated it. That particular constant wasn't available in the sword-server module, so I ended up using HttpStatus.SC_FORBIDDEN instead